test/exec/test_at_coroutine_exit.cpp

```diff
@@ -0,0 +1,307 @@
+#endif
+
+#ifdef STDEXEC_HAS_FORK
+namespace {
+template<class F, class... Args>
+void REQUIRE_TERMINATE(F&& f, Args&&... args)
+{
+  //spawn a new process
+  auto child_pid = ::fork();
+
+  //if the fork succeed
+  if (child_pid >= 0){
+
+    //if we are in the child process
+    if (child_pid == 0){
+
+        //call the function that we expect to abort
+        std::set_terminate([]{ std::exit(EXIT_FAILURE); });
+
+        std::invoke((F&&) f, (Args&&) args...);
+
+        //if the function didn't abort, we'll exit cleanly
+        std::exit(EXIT_SUCCESS);
+    }
+  }
+
+  //determine if the child process aborted
+  int exit_status{};
+  ::wait(&exit_status);
+
+  // we check the exit status instead of a signal interrupt, because
+  // Catch is going to catch the signal and exit with an error
+  bool aborted = WEXITSTATUS(exit_status);
+
+  REQUIRE(aborted);
+}
+}
+#endif
+
+using namespace exec;
+using stdexec::sync_wait;
+
+namespace {
+auto stop() {
+  static stopped_scheduler scheduler{};
+  auto stop = stdexec::schedule(scheduler);
+  return stop;
+}
+
+task<void> test_one_cleanup_action(int& result) {
+  ++result;
+  co_await at_coroutine_exit([&result]() -> task<void> { result *= 2; co_return; });
+  ++result;
+}
+
+task<void> test_two_cleanup_actions(int& result) {
+  ++result;
+  co_await at_coroutine_exit([&result]() -> task<void> { result *= 2; co_return; });
+  co_await at_coroutine_exit([&result]() -> task<void> { result *= result; co_return; });
+  ++result;
+}
+
+task<void> test_one_cleanup_action_with_stop(int& result) {
+  ++result;
+  co_await at_coroutine_exit([&result]() -> task<void> { result *= 2; co_return; });
+  co_await stop();
+  ++result;
+}
+
+task<void> test_two_cleanup_actions_with_stop(int& result) {
+  ++result;
+  co_await at_coroutine_exit([&result]() -> task<void> { result *= 2; co_return; });
+  co_await at_coroutine_exit([&result]() -> task<void> { result *= result; co_return; });
+  co_await stop();
+  ++result;
+}
+
+task<void> test_sender_cleanup_action(int& result) {
+  co_await at_coroutine_exit([&result]{ return stdexec::just() | stdexec::then([&result]{++result;}); });
+}
+
+task<void> test_stateful_cleanup_action(int& result, int arg) {
+  co_await at_coroutine_exit([arg,&result]{ return stdexec::just() | stdexec::then([arg,&result]{result += arg;}); });
+}
+
+task<void> test_mutable_stateful_cleanup_action(int& result) {
+  auto&& [i] = co_await at_coroutine_exit(
+    [&result](int&& i) -> task<void> {
+      result += i;
+      co_return;
+    }, 3);
+  ++result;
+  i *= i;
+}
+
+task<void> with_continuation(int& result, task<void> next) {
+  co_await std::move(next);
+  result *= 3;
+}
+
+void test_cancel_in_cleanup_action_causes_death(int& result) {
+  task<void> t = []() -> task<void> {
+    co_await at_coroutine_exit([]() -> task<void> {
+      co_await stop();
+    });
+  }();
+  sync_wait(std::move(t)); // causes termination
+  // ADD_FAILURE() << "He didn't fall? Inconceivable!";
```

Comment:
LOL nice. I'm curious why this is commented out though.

Reply:
Those tests are literally copied from unifex, which uses gtest, I think. I defined REQUIRE_TERMINATE to test for termination, although adding here a REQUIRE(false) is more accurate. I will add something equivalent.